src/fs/initial-project.ts

```diff
@@ -27,6 +29,21 @@ export interface InitialProject {
   isDefault: boolean;
 }
 
+export type MultiFilePythonProject = {
+  files: { [key: string]: string };
+  projectName?: string;
+};
+
+export const isSimplePythonProject = (
+  project: any
+): project is SimplePythonProject => {
+  return (
+    project &&
+    project.hasOwnProperty("main") &&
+    typeof project.main === "string"
+  );
+};
```

Comment:
Is this a format we need to be backwards compatible with?
V2 editor uses a string, I think?

Reply:
You are right, so we now receive a string or (`MultiFilePythonProject`) object from the host. If it's a string, it's resolved to this `SimplePythonProject` before being passed into the file system `resolve({ isDefault: false, main: data.projects[0] });`

Reply:
I've had a go at this here - [`7539a94`](https://github.com/microbit-foundation/python-editor-next/pull/626/commits/7539a94c8862618dcda7e690db1ddcaf3d5ebd3b). The `isDefault` field only seems to be used for testing, but I've kept it in for now.

Reply:
Ooh, I think isDefault was only used for the stats code we used the other day and can probably be removed now.

Reply:
Taken care of here - [`833c48a`](https://github.com/microbit-foundation/python-editor-next/pull/626/commits/833c48ad7f6adde6ef226f93025f9f59f3c3b27f)